analytics/pubstack/eventchannel/eventchannel.go

```diff
@@ -0,0 +1,146 @@
+package eventchannel
+
+import (
+	"bytes"
+	"compress/gzip"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+type Metrics struct {
+	bufferSize int64
+	eventCount int64
+}
+type Limit struct {
+	maxByteSize   int64
+	maxEventCount int64
+	maxTime       time.Duration
+}
+type EventChannel struct {
+	gz   *gzip.Writer
+	buff *bytes.Buffer
+
+	ch      chan []byte
+	endCh   chan int
+	metrics Metrics
+	mux     sync.Mutex
```

Comment:
Exactly which memory is being protected by the mux? It looks like the gz writier. Consider adding a comment to make this clear.

Reply:
I've renamed the mutex, which will be more readable everywhere